pkg/server/apiextensions.go

```diff
@@ -245,117 +259,226 @@ func (c *apiBindingAwareCRDLister) ListWithContext(ctx context.Context, selector
 	if err != nil {
 		return nil, err
 	}
 
-	if strings.HasSuffix(name, ".") {
-		name = name + "core"
+	if crd.Annotations != nil {
+		if identity := crd.Annotations[apisv1alpha1.AnnotationAPIIdentityKey]; identity != "" {
+			if updatedCRD.Annotations == nil {
+				updatedCRD.Annotations = make(map[string]string)
+			}
+			updatedCRD.Annotations[apisv1alpha1.AnnotationAPIIdentityKey] = identity
+		}
+
+		if _, partialMetadata := crd.Annotations[annotationKeyPartialMetadata]; partialMetadata {
+			updatedCRD = partialMetadataCRD(crd)
+		}
 	}
 
-	var crd *apiextensionsv1.CustomResourceDefinition
+	return updatedCRD, nil
+}
 
-	if cluster.Wildcard {
-		// HACK: Search for the right logical cluster hosting the given CRD when watching or listing with wildcards.
-		// This is a temporary fix for issue https://github.com/kcp-dev/kcp/issues/183: One cannot watch with wildcards
-		// (across logical clusters) if the CRD of the related API Resource hasn't been added in the root logical cluster first.
-		// The fix in this HACK is limited since the request will fail if 2 logical clusters contain CRDs for the same GVK
-		// with non-equal specs (especially non-equal schemas).
-		var crds []*apiextensionsv1.CustomResourceDefinition
-		crds, err = c.crdLister.List(labels.Everything())
-		if err != nil {
-			return nil, err
-		}
+// Get gets a CustomResourceDefinition.
+func (c *APIBindingAwareCRDLister) Get(ctx context.Context, name string) (*apiextensionsv1.CustomResourceDefinition, error) {
+	var (
+		crd *apiextensionsv1.CustomResourceDefinition
+		err error
+	)
 
-		// for PartialObjectMetadata we weaken the schema requirement and let requests through although the schemas differ
-		pruneEverythingButObjectMeta := false
-		if accept := ctx.Value(acceptHeaderContextKey).(string); len(accept) > 0 {
-			if _, params, err := mime.ParseMediaType(accept); err == nil {
-				pruneEverythingButObjectMeta = params["as"] == "PartialObjectMetadata" || params["as"] == "PartialObjectMetadataList"
-			}
-		}
+	clusterName, err := request.ClusterNameFrom(ctx)
+	if err != nil {
+		return nil, err
+	}
 
-		var crd *apiextensionsv1.CustomResourceDefinition
-		if pruneEverythingButObjectMeta {
-			crd := findCRDIgnoringSpec(name, crds)
-			if crd != nil {
-				crd := crd.DeepCopy()
-				// set minimal schema that prunes everything but ObjectMeta
-				for _, v := range crd.Spec.Versions {
-					v.Schema = &apiextensionsv1.CustomResourceValidation{
-						OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
-							Type: "object",
-						},
-					}
-				}
-				return crd, nil
-			}
-		} else {
-			var equal bool // true if all the found CRDs have the same spec
-			crd, equal = findCRD(name, crds)
-			if !equal {
-				err = apierrors.NewInternalError(fmt.Errorf("error resolving resource: cannot watch across logical clusters for a resource type with several distinct schemas"))
-				return nil, err
-			}
-		}
+	if strings.Contains(name, "kcp.dev") {
```

Comment:
this means we have different concepts of system CRDs. Can't we call `getSystemCRD` and fall through on IsNotFound?

Reply:
k